java/src/main/java/com/bacheconsulting/kata/encode/ByteBuffer.java

```diff
@@ -1,24 +1,22 @@
-package encodeKata;
+package com.bacheconsulting.kata.encode;
 
 import java.util.ArrayList;
 import java.util.List;
 
 public class ByteBuffer {
-    private final List<Integer> buffer = new ArrayList<>();
+
+    private final List<Byte> buffer = new ArrayList<>();
```

Comment:
I'm curious why you changed this from Integer to Byte? The original C++ code uses an integer. I wasn't sure what the difference is?

Reply:
Although I can live with method arguments and return values to be `int` even if the real value can just contain 8 valid bits (there are reasons in the Java world dictated by early design decisions which weren't lucky but we have to live with them), the code purist in me shouts **"NO"** when I see a class named `ByteBuffer` with the underlying collection to be defined as `List<Integer>`. 😄

Although in reality, all the _bytes_ in the collection will be allocated as _objects_ and not primitives (so the memory footprint would be very similar regardless of my changes), I like the expressivity of `List<Byte>` over `List<Integer>`, declaring that we're really storing just bytes.

Reply:
ok! Thanks for the explanation. Bytes is fine :-)